File: w1_p4/30.c
#include <stdio.h>
int main() {
	char m[100][101];
	int i,j,q,w,e,r,n;
	int p=1;
	
	while(scanf("%d %d",&i,&j) == 2)
	{
		
		if(i==0 && j==0)
		{
			break;
		}
		
		printf("Field #%d:\n",p);
		p++;
		
		for(q=0; q<i; q++)
		{
			for(w=0; w<j; w++)
			{
				scanf(" %c",&m[q][w]);
			}
		}
	
		
		for(e=0; e<i; e++)
		{
			for(r=0; r<j; r++)
			{
			n=0;
			
				if(m[e][r]=='*')
				{
					printf("%c",m[e][r]);
					continue;
				}

				if(e-1>=0)
				{
					
					if(m[e-1][r]=='*')
					{
						n++;
					}
					
					if(r-1>=0)
					{
						if(m[e-1][r-1]=='*')
						{
							n++;
						}	
					}

					if(r+1<j)
					{
						if(m[e-1][r+1]=='*')
						{
							n++;
						}
					}
				}
				
				if(e+1<i)
				{
					if(m[e+1][r]=='*')
					{
						n++;
					}
					
					if(r-1>=0)
					{
						if(m[e+1][r-1]=='*')
						{
							n++;
						}
					}
					
					if(r+1<j)
					{
						if(m[e+1][r+1]=='*')
						{
							n++;
						}
					}
					
				}
				
					if(r-1>=0)
					{
						if(m[e][r-1]=='*')
						{
							n++;
						}
					}
				
					if(r+1<j)
					{
						if(m[e][r+1]=='*')
						{
							n++;
						}
					}
				
			printf("%d",n);
				//printf("%c",m[0][0]);
			}
			printf("\n");
		}
		
		
		printf("\n");
		
	}
	return 0;
}

File: w1_p4/30.java
import java.io.*;
class Main {
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		String input;
		int cnt = 1;
		while ((input = br.readLine()) != null) {
			String[] num = input.split(" ");
			int m = Integer.parseInt(num[0]);
			int n = Integer.parseInt(num[1]);
			
			if (m == 0 && n == 0) {
				break;
			} else {
				char[][] chara = new char[m][n];
				for (int i = 0; i < m; i++) {
					chara[i] = br.readLine().toCharArray();
				}
				int[][] arr = new int[m][n];
				for (int i = 0; i < m; i++) {
					for (int j = 0; j < n; j++) {
						if (chara[i][j] == '*') {
							add(arr, i, j);
						}
					}
				}
				System.out.println("Field #" + cnt++ + ":");
				for (int i = 0; i < m; i++) {
					for (int j = 0; j < n; j++) {
						if (chara[i][j] == '*') {
							System.out.print(chara[i][j]);
						} else {
							System.out.print(arr[i][j]);
						}
					}
					System.out.println();
				}
				System.out.println();
			}
		}
	}
	
	static void add(int[][] arr, int m, int n) {
		for (int i = m - 1; i <= m + 1; i++) {
			for (int j = n - 1; j <= n + 1; j++) {
				try {
					arr[i][j]++;
				} catch (ArrayIndexOutOfBoundsException e) {
					continue;
				}
			}
		}
	}
}